includes/multi-currency/MultiCurrency.php

```diff
@@ -1441,28 +1441,55 @@ public function is_multi_currency_settings_page(): bool {
 		);
 	}
 
+	/**
+	 * Function used to compute the customer used currencies, used as internal callable for get_all_customer_currencies function.
+	 *
+	 * @return array
+	 */
+	public function callable_get_customer_currencies() {
+		global $wpdb;
+
+		$currencies  = $this->get_available_currencies();
+		$query_union = [];
+
+		if ( class_exists( 'Automattic\WooCommerce\Utilities\OrderUtil' ) &&
+					\Automattic\WooCommerce\Utilities\OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			foreach ( $currencies as $currency ) {
+				$query_union[] = sprintf(
+					'SELECT "%1$s" AS currency_code, EXISTS(SELECT currency FROM %2$s WHERE currency="%1$s" LIMIT 1) AS exists_in_orders',
+					$currency->code,
+					$wpdb->prefix . 'wc_orders'
+				);
+			}
+		} else {
+			foreach ( $currencies as $currency ) {
+				$query_union[] = sprintf(
+					'SELECT "%1$s" AS currency_code, EXISTS(SELECT meta_value FROM %2$s WHERE meta_key="_order_currency" AND meta_value="%1$s" LIMIT 1) AS exists_in_orders',
```

Comment:
@ovidiul, similar question here, did we check with `EXPLAIN` which indexes are used in this query? Can you share perhaps a sample output?

Reply:
Here is the explain of the generated query

```
+-----+-------------+--------------+------------+------+---------------+-------------+---------+-------+------+----------+-----------------------------+
| id  | select_type | table        | partitions | type | possible_keys | key         | key_len | ref   | rows | filtered | Extra                       |
+-----+-------------+--------------+------------+------+---------------+-------------+---------+-------+------+----------+-----------------------------+
|   1 | PRIMARY     | <derived2>   | NULL       | ref  | <auto_key0>   | <auto_key0> | 8       | const |   10 |   100.00 | Using where; Using filesort |
|   2 | DERIVED     | NULL         | NULL       | NULL | NULL          | NULL        | NULL    | NULL  | NULL |     NULL | No tables used              |
|   3 | SUBQUERY    | wp2_postmeta | NULL       | ref  | meta_key      | meta_key    | 767     | const |    1 |    16.67 | Using where                 |
|   4 | UNION       | NULL         | NULL       | NULL | NULL          | NULL        | NULL    | NULL  | NULL |     NULL | No tables used              |
...
```

using the meta_key as index.

Reply:
Blazing fast, thanks for sharing details!